### ISSUE TITLE
I want to be able to store my local data

### ISSUE THREAD

**Issue by christianbrb** [raiden-network/light-client#152]
## User Story
As Chris, I want to be able to store my local data so that I can restart in another browser or on another machine.

## Acceptance criteria
- JSON is locally downloaded
- JSON can be loaded back again
- User knows they shouldn't run two instances in parallel without syncing their state by downloading/uploading the respective states before starting each instance

## Tasks
- [ ] Split issue and iterate
- [ ] Initial UX/UI design (download/upload button)
- [ ] Initial soultion, add a download button, alt. dropdown
- [ ] "Proper" solution, introduce the settings menu

**Comment by christianbrb** [comment]
I have met @RachBLondon from [3Box](https://3box.io/). They have been developing a wrapper to easily use IPFS.
MetaMask is also using it to enable the user to [backup their data](https://medium.com/3box/how-metamask-backs-up-wallet-data-using-3box-storage-c90a34f9e56f).

**Comment by RachBLondon** [comment]
@christianbrb happy to set up a call if thats useful

**Comment by christianbrb** [comment]
We check with Rachel how easy it is to set this up and create a new issue for the 3Box integration.

First step: 
Just download and upload the JSON locally -> we would need a UI for it -> Settings Screen?

Upload:
- Stop the SDK
- Block the UI
- Show a snackbar

Prototype: Web-UI Address Book

**Comment by andrevmatos** [comment]
This depends on #614

**Comment by andrevmatos** [comment]
Ok, now that the above issue is solved, this can be implemented:
- To download the state, there may be a Save State button in the **running** instance of the dApp:
  - Once the user clicks to download, warns them that the dApp will be stopped, and that they should only connect to it again on browser if they don't plan to load the Saved State somewhere else, as running multiple instances of the Light Client in parallel is not supported and may cause undefined behavior
  - Stop the SDK, call `encodeRaidenState(raiden.state)` (exported from `raiden-ts`, which will return a stringified JSON), and offer it to be downloaded as a `JSON` file.
    - Filename suggestion: `raiden_goerli_${address}_2020-02-18T11:44:13.json`
- To upload the state, there may be a Load State button in the **stopped** instance, aka Connect screen:
  - The file content may be passed as a string, the SDK will parse it. No need to validate filename, SDK will validate its content
  - Loading a state works as clicking on `Connect`, but with the new state, therefore it's expected that the account on Metamask is loaded and matches the one in the state. It'll reject if not.
  - If the user has used this account on current browser **after** the backup (meaning there's a stored state which is newer than the backup), it'll reject as well. The user will need to clean `localStorage` by hand then if wants to force loading the backup.
  - The SDK create call can be called as:
```ts
const raiden = await Raiden.create(provider, account, { storage: window.localStorage, state: 'string' }, ...);
```

**Comment by taleldayekh** [comment]
Currently looking at an initial design solution for this.

**Comment by taleldayekh** [comment]
![general](https://user-images.githubusercontent.com/43838780/75900330-0d74dd00-5e3d-11ea-91a1-bd351b1db6c6.jpg)
